fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+        The string-keyword indicates a
+        distribution type, see user guide for more information.
+        A callable should be a
+        loss function with an API that follows the chosen backend (torch or
+        tensorflow).
+
+    predictor_function : str, callable, default = 'auto'
+        Either the string
+        :code:`'auto'`, :code:`'binary'`, :code:`'category'`,
+        :code:`'continuous'`, or a callable. The string
+        :code:`'auto'` indicates to infer the predictor function
+        from the distribution type of :code:`y`
+        The string-keyword indicates a
+        distribution type, see user guide for more information.
+        A callable should be a
+        function that maps the continuous output of the predictor model to
+        a discrete prediction.
+
+    predictor_optimizer : str, torch.optim, tensorflow.keras.optimizers, callable, default = 'Adam'
+        The optimizer class to use. If a string is passed instead, this must be
+        either "SGD" or "Adam". A corresponding SGD or Adam optimizer is
+        initialized with the model and given learning rate.
+        If an instance of a subclass of torch.optim.Optimizer
+        or tensorflow.keras.optimizers.Optimizer is passed, this
+        is used directly. If a callable :code:`fn` is passed, we call this
+        callable and pass our model, and set the result of this call
+        as the optimizer, so: :code:`predictor_optimizer=fn(predictor_model)`.
+
+
+    adversary_optimizer : str, torch.optim, tensorflow.keras.optimizers, callable, default = 'Adam'
+        The optimizer class to use. Similarly defined as
+        :code:`predictor_optimizer`
+
+    constraints : str, default = 'demographic_parity'
+        The fairness measure to optimize for. Must be either 'demographic_parity'
+        (Demographic Parity) or 'equalized_odds' (Equalized Odds).
+
+    y_transform : sklearn.base.TransformerMixin, default = fairlearn.adversarial.FloatTransformer("auto")
+        The preprocessor to use on the predictions :code:`y`.
+        The given transformer *must* map data
+        to a 2d ndarray containing only floats. Per default, we use a
+        FloatTransformer that maps strictly binary or categorical tables
+        to binary or
+        one-hot encodings, and it maps strictly continuous-valued (possible 2d)
+        to itself.
+
+    z_transform : sklearn.base.TransformerMixin, default = fairlearn.adversarial.FloatTransformer("auto")
+        The preprocessor to use on the :code:`sensitive_features`.
+        The given transformer *must* map data
+        to a 2d ndarray containing only floats. Per default, we use a
+        FloatTransformer that maps strictly binary or categorical tables
+        to binary or
+        one-hot encodings, and it maps strictly continuous-valued (possible 2d)
+        to itself.
+
+    learning_rate : float, default = 0.001
+        A small number greater than zero to set as initial learning rate
```

Comment:
Should the parameter be called `initial_learning_rate` then?

Reply:
it depends on what the user does with the optimizers, decaying learning rate is not the default (though it would be nice!). About the optimizers, I'm not entirely a fan of my own work here, as initializing the optimizers is quite messy. But despite trying I did not see a more elegant solution that provides flexibility to the user. Maybe you have more experience with this?

Reply:
oh sorry, I see now that I wrote "initial" in the sentence below